src/tasks/addonindex/taskaddonindex.cpp

```diff
@@ -43,15 +45,18 @@ void TaskAddonIndex::run() {
   }
 
   // Index file signature
-  if (Feature::get(Feature::Feature_addonSignature)->isSupported()) {
-    NetworkRequest* request = NetworkRequest::createForGetUrl(
-        this,
-        QString("%1manifest.json.sig").arg(AddonManager::addonServerAddress()),
-        200);
+  if (AddonManager::signatureVerificationNeeded()) {
+    QString manifestSigUrl =
+        QString("%1manifest.json.sig").arg(AddonManager::addonServerAddress());
+    logger.debug() << "Download manifest signature URL:" << manifestSigUrl;
+
+    NetworkRequest* request =
+        NetworkRequest::createForGetUrl(this, manifestSigUrl, 200);
 
     connect(request, &NetworkRequest::requestFailed, this,
             [this](QNetworkReply::NetworkError error, const QByteArray&) {
               logger.error() << "Get addon index signature failed" << error;
+              AddonManager::instance()->updateIndex(false);
```

Comment:
Why is this necessary if the index did not change?

Reply:
to communicate the end of the operation to tests. If we don't do that, we will never know if the current operation succeeded or not.